articles/guides/login/migrating-lock-v10-spa.md

```diff
@@ -0,0 +1,105 @@
+---
+title: Moving SPAs using Lock to Centralized Login 
+description: Learn how to migrate from Single Page Applications using Lock to Centralized Login
+toc: true
+---
+
+# Migrate SPAs using Lock 10+ to Centralized Login
+
+This document explains how to migrate Single Page Applications using Lock to centralized login. 
+
+For other migration scenarios see [Migrating from Embedded to Centralized Login](/guides/login/migration-embedded-centralized).
+
+When you use Lock, your code does basically this:
+
+1. Initialize Lock:
+
+```js
+var lock = new Auth0Lock('${account.clientId}', '${account.namespace}', {
+    auth: {
+      redirectUrl: '${account.callback}',
+      responseType: 'token id_token',
+      audience: 'https://' + '${account.namespace}' + '/userinfo',
+      params: {
+        scope: 'openid'
+      }
+    }
+});
+```
+
+2. Set the session and update the UI in the `authenticated` event:
+ 
+```js
+lock.on('authenticated', function (authResult) {
+    if (authResult && authResult.accessToken && authResult.idToken) {
+      setSession(authResult); // set the local session
+    }
+    displayButtons(); // refresh the UI to indicated a ‘logged-in’ state.
+});
+```
+
+3. Handle errors in the `authorization_error` event:
+
+```js
+// Handle authorization errors
+lock.on('authorization_error', function (err) {
+    console.log(err);
+    alert('Error: ' + err.error + '. Check the console for further details.');
+    displayButtons();
+});
+```
+
+4. Show lock when the login button is clicked:
```

Comment:
I'd change it to "Show Lock when a login is required" (or triggered), because it could be a login button but also the user trying to access a protected resource.